libs/pulumi-lambda/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,289 @@
+                        },
+                    },
+                    {
+                        id: 'invocations',
+                        label: 'Invocations',
+                        metric: {
+                            namespace: 'AWS/Lambda',
+                            metricName: 'Invocations',
+                            stat: 'Sum',
+                            dimensions: {
+                                FunctionName: args.lambdaFunctionName,
+                                Resource: args.lambdaFunctionName,
+                            },
+                            period: 60,
+                        },
+                    },
+                ],
+                evaluationPeriods: 2,
+                datapointsToAlarm: 2,
+                comparisonOperator: 'GreaterThanOrEqualToThreshold',
+                threshold:
+                    args.thresholds?.errorRatePercent ??
+                    defaultThresholds.errorRatePercent,
+                alarmDescription: pulumi.interpolate`threshold: error rate >= ${
+                    args.thresholds?.errorRatePercent ??
+                    defaultThresholds.errorRatePercent
+                }%: ${args.lambdaFunctionName}`,
+                alarmActions: [args.snsTopicArn],
+                okActions: [args.snsTopicArn],
+                insufficientDataActions: [args.snsTopicArn],
+                treatMissingData: 'notBreaching',
+            },
+            {
+                parent: this,
+                deleteBeforeReplace: true,
+            },
+        )
+
+        new aws.cloudwatch.MetricAlarm(
+            `${name}-duration-avg`,
+            {
+                namespace: 'AWS/Lambda',
+                metricName: 'Duration',
+                statistic: 'Average',
+                period: 60,
+                evaluationPeriods: 2,
+                datapointsToAlarm: 2,
+                comparisonOperator: 'GreaterThanOrEqualToThreshold',
+                threshold:
+                    args.thresholds?.avgDurationMs ??
+                    defaultThresholds.avgDurationMs,
+                alarmDescription: pulumi.interpolate`threshold: avg duration >= ${
+                    args.thresholds?.avgDurationMs ??
+                    defaultThresholds.avgDurationMs
+                }: ${args.lambdaFunctionName}`,
+                dimensions: {
+                    FunctionName: args.lambdaFunctionName,
+                    Resource: args.lambdaFunctionName,
+                },
+                alarmActions: [args.snsTopicArn],
+                okActions: [args.snsTopicArn],
+                insufficientDataActions: [args.snsTopicArn],
+                treatMissingData: 'missing',
+            },
+            {
+                parent: this,
+                deleteBeforeReplace: true,
+            },
+        )
+
+        if (args.thresholds?.maxDurationMs) {
+            new aws.cloudwatch.MetricAlarm(
+                `${name}-duration-max`,
```

Comment:
Whats the difference between durationMax and timeout?

Reply:
Just the alert that's raised. It should be a bit clearer now that it's split up